pkg/controller/operators/catalog/operator.go

```diff
@@ -1654,9 +1693,58 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				plan.Status.Plan[i].Status = status
 
 			default:
-				return v1alpha1.ErrInvalidInstallPlan
-			}
+				if !isSupported(step.Resource.Kind) {
+					// Not a supported resource
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusUnsupportedResource
+					return v1alpha1.ErrInvalidInstallPlan
+				}
+
+				// Marshal the manifest into an unstructured object
+				dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(step.Resource.Manifest), 10)
+				unstructuredObject := &unstructured.Unstructured{}
+				if err := dec.Decode(unstructuredObject); err != nil {
+					return errorwrap.Wrapf(err, "error decoding %s object to an unstructured object", step.Resource.Name)
+				}
+
+				// Get the resource from the GVK.
+				gvk := unstructuredObject.GroupVersionKind()
+				r, err := o.apiresourceFromGVK(gvk)
+				if err != nil {
+					return err
+				}
+
+				// Create the GVR
+				gvr := schema.GroupVersionResource{
+					Group:    gvk.Group,
+					Version:  gvk.Version,
+					Resource: r.Name,
+				}
+
+				// Set up the dynamic client ResourceInterface
+				var resourceInterface dynamic.ResourceInterface
+				if r.Namespaced {
+					ownerutil.AddOwner(unstructuredObject, plan, false, false)
+					unstructuredObject.SetNamespace(namespace)
+					resourceInterface = o.dynamicClient.Resource(gvr).Namespace(namespace)
+				} else {
+					resourceInterface = o.dynamicClient.Resource(gvr)
+				}
```

Comment:
This seems like a lot of ceremony just to create an unstructured resource. I think controller-runtime's client handles this for you.

Reply:
Controller-runtime's client does not require the GVR - something we could switch to but it might make more sense to rip the dynamic-client out entirely given that the dynamic client is used elsewhere in OLM.